### ISSUE TITLE
Uncaught TypeError: Cannot destructure property 'A' of '_ember.default' as it is undefined.

### ISSUE THREAD

**Issue by liberlanco** [emberjs/ember-inspector#1595]
**Describe the bug**
Exception in console: `Uncaught TypeError: Cannot destructure property 'A' of '_ember.default' as it is undefined.`

From time to time Ember also said that this is not an Ember application. I suppose because of this is error.

With more loaded application  (tailwind,  ember-cli-mirage, some components, some routes, some models and so on) it starts failing more often. When I did mirage mockups, and gives this error in 50-60% of refreshes.

**To Reproduce**
1. Install ember-cli 3.26.1,  and ember-inspector 4.4.0 in Google Chrome browser
2. Create new project and start it: `mkdir tmp-dir;  cd tmp-dir;  ember init;  npm start`
3. Open http://127.0.0.1:4200/  and manually add some non-existent url like http://127.0.0.1:4200/some/url
4. Open chrome inspector dev tools (F12)
5. Open console
6. Now spam Ctrl-R to refresh page. After time to time there will be exception:
```
VM4178:61 Uncaught TypeError: Cannot destructure property 'A' of '_ember.default' as it is undefined.
    at <anonymous>:61:5
    at requireModule (<anonymous>:42:28)
    at requireModule (<anonymous>:38:24)
    at <anonymous>:9574:62
    at triggerOnce (<anonymous>:9642:7)
    at runLoadHooks (lazy_load.js:75)
    at index.js:588
    at internalRequire (loader.js:54)
    at require (loader.js:87)
    at rsvp.js:2493
```

Research showed that it points to this line: https://github.com/emberjs/ember-inspector/blob/master/ember_debug/adapters/basic.js#L3   Ember object here could be undefined.

**Expected behavior**
No exception

**Environment**
ember 3.26.1
node-inspector 4.4.0
google chrome  90.0.4430.212

**Comment by rwwagner90** [comment]
@pzuraq @chancancode is there a good way to ensure Ember is defined here? This seems to be related to switching to using `require` instead of the Ember global.

**Comment by pzuraq** [comment]
@rwwagner90 I'm not exactly sure how that could be `undefined`, it's either loaded via the module or via `window.Ember`: https://github.com/emberjs/ember-inspector/blob/master/ember_debug/utils/ember.js

The only way this seems like it could cause this error is if `window.Ember` is `undefined`, so it's not an Ember application at all. In that case, we may need better logic to prevent the inspector from loading. 

If this is happening on actual Ember applications where `window.Ember` is defined, can we get a reproduction?

**Comment by pzuraq** [comment]
Oh, possibly what's happening here is inspector is initializing itself before Ember has had a chance to add the global, there could be a race condition and that's why it only happens sometimes

**Comment by liberlanco** [comment]
Yes, looks like race condition.  window.Ember is defined most of the refreshes. And even when I see this exception, I check window.Ember and it's defined.  

I can reproduce it even on empty app with Welcome Page, as described in steps to reproduce, but after 10-20 attempts.  On my real app it happens more often, so probably adding some components, some models, and plugins increase chance to reproduce.   May be also opening more tabs with an application also could increase the chance.

**Comment by tibomogul** [comment]
Happening to my team also, just today. Does not happen to a  teammate that is still on 4.3.5.

**Comment by dirkdirk** [comment]
Can someone please post the Chrome Extension ID for Ember Inspector 4.3.5 so we can downgrade in extension?

**Comment by jayvarner** [comment]
@dirkdirk I putzed around on this a bit this morning. I resolved it by making a few changes to the instructions on the readme:
 
- Remove the Ember Inspector extension from Chrome
- Clone the repository
- cd into the repo directory
- switch to the 4.3.5 tag by running `git checkout tags/v4.3.5`
- run `yarn install`
- run `yarn global add ember-cli`
- run `yarn build` to build the `dist` directory
- Visit `chrome://extensions` in Chrome
- Make sure `Developer mode` is checked
- Click on 'Load unpacked extension...'
- Choose the `dist/chrome` folder in the cloned repo
- Close and re-open developer tools if it's already open

I hope that helps

**Comment by dirkdirk** [comment]
@jayvarner ... impressive!

On your machine after doing magik, when you go to the Extension Manager, with Dev Mode checked, does it give the ID for 4.3.5?

**Comment by jayvarner** [comment]
ijbpeeflkojiljfgfdkhbajcombijegb

**Comment by Rabbi50** [comment]
Happening to me also from today. How to solve it please anyone help.

**Comment by dirkdirk** [comment]
> Happening to me also from today. How to solve it please anyone help.

The best options as of now are:

1. Turn off the Chrome Ember Inspector extension.
2. Uninstall the Chrome Ember Inspector extension.
3. Downgrade following jayvarner's directions above.

**Comment by efx** [comment]
## alternate workaround

1. In Google Chrome, navigate to the Ember Inspector extension
1. Toggle on "Collect errors" 

## debugging notes
If I inspect the generated background page I see the following error from the extension:
```js
_generated_background_page.html:1 Error in event handler: TypeError: Error in invocation of tabs.sendMessage(integer tabId, any message, optional object options, optional function responseCallback): No matching signature.
    at chrome-extension://bmdblncegkenkacieihfhpjfppoconhi/background-script.js:139:21
```
I'm unsure if that matters or is a false positive.

**Comment by rwjblue** [comment]
We just tagged v4.4.1 with a fix (though there may need to be more fixes to follow).

**Comment by rwjblue** [comment]
I believe that version is now published to the chrome store.

**Comment by rwwagner90** [comment]
This should be fixed as of 4.4.1. If anyone is on 4.4.1 and still seeing this, please let us know!
